File: src/main/java/com/refinedmods/refinedstorage/api/util/Action.java
package com.refinedmods.refinedstorage.api.util;

/**
 * Defines how an action is performed.
 */
public enum Action {
    /**
     * Performs the action.
     */
    PERFORM,
    /**
     * Gives back the same return as called with PERFORM, but doesn't mutate the underlying structure.
     */
    SIMULATE
}

File: src/main/kotlin/com/refinedmods/refinedstorage/api/util/Action.kt
package com.refinedmods.refinedstorage.api.util



/**
 * Defines how an action is performed.
 */
enum class Action {
    /**
     * Performs the action.
     */
    PERFORM,

    /**
     * Gives back the same return as called with PERFORM, but doesn't mutate the underlying structure.
     */
    SIMULATE
}